src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -207,7 +207,30 @@ public ControlTemplate Template
 
 		private protected virtual void OnTemplateChanged(DependencyPropertyChangedEventArgs e)
 		{
-#if !UNO_HAS_ENHANCED_LIFECYCLE
+#if UNO_HAS_ENHANCED_LIFECYCLE
```

Comment:
In theory, adding a template will automatically incur a measure invalidation, since it's adding a frameworkelement to the tree that will itself request a measure.